tower/src/ready_cache/cache.rs

```diff
@@ -193,7 +193,7 @@ where
     }
 
     /// Obtains a mutable reference to a service in the ready set by index.
-    pub fn get_ready_index_mut(&mut self, idx: usize) -> Option<(&mut K, &mut S)> {
+    pub fn get_ready_index_mut(&mut self, idx: usize) -> Option<(&K, &mut S)> {
```

Comment:
This is a breaking change, so I suppose it should go on your 0.5 milestone. But if you *really* want the mutable key, there's an opt-in by using the `MutableKeys` trait and its [`get_index_mut2`](https://docs.rs/indexmap/2.0.0/indexmap/map/trait.MutableKeys.html#tymethod.get_index_mut2) method.

Reply:
It's feels like more of a footgun to have the key be mutable, given the warnings mentioned in `MutableKeys::get_index_mut2`... and there's really nothing in the original PR or docs that speaks specifically to the _key_ needing to be mutable... so I think this is a worthwhile change to make given that changing the MSRV is breaking anyways.